### PR TITLE
Add Habitat support

### PR DESCRIPTION
**Pull request by smith**
This makes it so you can scan habitat packages and their dependencies.
It should work, however, it's only going to have `license_files` if a
package contains a file called LICENSE (see habitat-sh/habitat#1738),
which most currently don't.

This is only a first step, and does not take into account "internal"
dependencies, like if a package is built for a Node or Ruby project.
This kind of stuff can be built into the build process for Habitat
packages.

Signed-off-by: Nathan L Smith <smith@chef.io>

### CODE REVIEW

**Comment by smith** [comment]
I need to fix some tests (stub out some files that I had locally but aren't there in Travis), but should be good once that's done.

File: lib/license_scout/dependency_manager/habitat.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+          end
+        end
+
+        extend Forwardable
+
+        attr_reader :path
+        def_delegators :ident, :origin, :name, :version, :release
+        def_delegators :manifest, :license
+
+        HAB_FS_ROOT = "/hab/pkgs"
```
Comment (danielsdeleo) [inline lib/license_scout/dependency_manager/habitat.rb]: `.freeze`

**Comment by danielsdeleo** [comment]
LGTM when tests are passing

**Comment by sersut** [comment]
Code wise 👍 . 

One thing to be cautious of is that when this code detects a component and sets the `license_files` for the component to `[ ]` some of the builds will fail (including automate) because they are set to fail on missing licensing information. 

This can be mitigated by providing overrides, relative paths or public urls but we need to include some code like this to make use of this functionality:

https://github.com/chef/license_scout/blob/master/lib/license_scout/dependency_manager/berkshelf.rb#L70-L79

I would try an automate build with this before merging to make sure we don't break the build.

**Comment by smith** [comment]
> One thing to be cautious of is that when this code detects a component and sets the license_files for the component to [ ] some of the builds will fail (including automate) because they are set to fail on missing licensing information.

@sersut We aren't currently running license scout against habitat stuff in automate, so I don't think it will affect the existing build process.

We will want to get this taken care of before we do, but this addition is just to add initial support that can be built out more fully in the future.

**Comment by smith** [comment]
@danielsdeleo @sersut updated with your suggested changes and working specs. Will merge once green if it's ok with you.

**Pull request merged**